src/filesys.cpp

```diff
@@ -740,5 +740,26 @@ bool Rename(const std::string &from, const std::string &to)
 	return rename(from.c_str(), to.c_str()) == 0;
 }
 
+bool bad_char(const char c){
+	if (c >= 0 && c <= 31)
+		return true;
+	if (bad_chars.find_first_of(c) != std::string::npos)
+		return true;
+	if (c > 255)
+		return true;
+	return false;
+}
+
+void FixBadFilename(std::string &input){
+	input.erase(remove_if(input.begin(), input.end(), bad_char), input.end());
+	input.erase(input.find_last_not_of(" ") + 1);
+	for (unsigned int a = 0; a < sizeof(bad_names) / sizeof(bad_names[0]); a++){
+		if (input == bad_names[a]){
```

Comment:
It compares each index of bad_names with the input, so it does what it's supposed to do.